R/createParameters.R

```diff
@@ -243,52 +243,57 @@ createParameters <- function(m, config, inputDir) {
   # Calculate share of buildings that need to be renovated or demolished between
   # given time steps assuming a Weibull distribution of the technology life time.
   # When passing the standing life time, the share of the initial stock standing
-  # in ttot2 that has to be demolished or renovated until given time step is
+  # in ttotIn that has to be demolished or renovated until given time step is
   # calculated
-  shareRen <- function(ttot2, params, standingLifeTime = NULL) {
+  shareRen <- function(params, standingLifeTime = NULL, timePeriods = NULL) {
 
-    share <- expandSets(ttot2 = "ttot", "ttot", .m = m) %>%
-      filter(.data$ttot2 <= .data$ttot) %>%
-      left_join(readSymbol(p_dt) %>%
-                  rename(dt = "value"),
-                by = c(ttot2 = "ttot")) %>%
+    if (is.null(timePeriods)) {
+      timePeriods <- expandSets(ttotIn = "ttot", ttotOut = "ttot", .m = m) %>%
+        filter(.data$ttotIn <= .data$ttotOut)
+    }
+
+    share <- timePeriods %>%
       cross_join(params) %>%
       pivot_wider(names_from = "variable")
 
     share <- if (is.null(standingLifeTime)) {
-      # average past flow activity happened dt/2 before nominal time step ttot2
+      # average past flow activity happened dt/2 before nominal time step ttotIn
       share %>%
-        mutate(lt = .data$ttot - (.data$ttot2 - .data$dt / 2),
+        left_join(readSymbol(p_dt) %>%
+                    rename(dt = "value"),
+                  by = c(ttotIn = "ttot")) %>%
+        filter(!is.na(.data$dt)) %>%
+        mutate(lt = .data$ttotOut - (.data$ttotIn - .data$dt / 2),
                p0 = 0)
     } else {
       # standing life time considers stock not flows -> no consideration of dt
       share %>%
-        mutate(lt = .data$ttot - .data$ttot2 + standingLifeTime,
+        mutate(lt = .data$ttotOut - .data$ttotIn + standingLifeTime,
                p0 = pweibull(standingLifeTime, .data$shape, .data$scale))
     }
 
     share %>%
-      mutate(p  = pweibull(.data$lt, .data$shape, .data$scale),
-             value = (.data$p - .data$p0) / (1 - .data$p0),
+      mutate(p  = pweibull(.data$lt, .data$shape, .data$scale)) %>%
+      mutate(value = (.data$p - .data$p0) / (1 - .data$p0),
              value = ifelse(.data$value > cutOffShare, 1, .data$value)) %>%
-      select(-"shape", -"scale", -"dt", -"lt", -"p", -"p0")
+      select(-any_of(c("shape", "scale", "dt", "lt", "p", "p0")))
   }
 
   ## building ====
 
   # parameter for monitoring purposes
-  p_probDem <- expandSets("region", "typ", ttot2 = "ttot", "ttot", .m = m) %>%
-    filter(.data[["ttot2"]] >= .data[["ttot"]]) %>%
+  p_probDem <- expandSets("region", "typ", ttotIn = "ttot", ttotOut = "ttot", .m = m) %>%
+    filter(.data[["ttotOut"]] >= .data[["ttotIn"]]) %>%
```

Comment:
Very minor: If you want to, you could change `.data[["..."]]` to `.data$...` where you change code anyways...